ui/app/AppLayouts/Wallet/popups/ReceiveModal.qml

```diff
@@ -23,16 +23,21 @@ StatusModal {
     id: root
 
     property string address: RootStore.selectedReceiveAccount.address
-    property string chainShortNames: ""
+    property string chainShortNames: RootStore.getNetworkShortNames(d.preferredSharingNetworksString)
+    property var preferredSharingNetworksArray: d.preferredSharingNetworksString.split(":").filter(Boolean)
 
     property string description: qsTr("Your Address")
 
     property bool readOnly: false
 
     QtObject {
         id: d
-        property string completeAddressWithNetworkPrefix
-        property var preferredSharingNetworksArray: !!RootStore.selectedReceiveAccount ? RootStore.selectedReceiveAccount.preferredSharingChainIds.split(":").filter(Boolean): []
+        property string completeAddressWithNetworkPrefix: root.chainShortNames + root.address
+        property string preferredSharingNetworksString: !!RootStore.selectedReceiveAccount ? RootStore.selectedReceiveAccount.preferredSharingChainIds : ""
+        onPreferredSharingNetworksStringChanged: {
```

Comment:
I think that you don't need `onPreferredSharingNetworksStringChanged` slot, cause because of bindings  `chainShortNames` and `preferredSharingNetworksArray` will be reevaluated.

Reply:
the problem is it si nThe binding is broken when I reassign to it on line below and hence wont work